### DISCUSSION TITLE
Use Stateless Client for administrative app

### DISCUSSION THREAD

**Discussion by stopyellingatme** [supabase-community/supabase-csharp#114] [Q&A] [answered]
There are instances when a user needs to be signed up without updating the session storage (admin panel generating users).

if an option could be added to the SignUpOptions.cs file and a conditional added to the Client.cs file, then I think that would fix the issue.

<img width="401" alt="Screenshot 2023-10-02 at 22 40 23" src="https://github.com/supabase-community/supabase-csharp/assets/1787881/ef69d72c-b6d4-4142-9129-96faaefb34b4">
<img width="679" alt="Screenshot 2023-10-02 at 22 41 17" src="https://github.com/supabase-community/supabase-csharp/assets/1787881/720483a1-6aba-4cb8-9743-58fee973cfd7">

**Comment by wiverson** [comment]
Hmm, maybe check out https://github.com/supabase-community/gotrue-csharp/blob/master/Gotrue/StatelessClient.cs#L27 and see if that does the trick?

**Comment by stopyellingatme** [comment]
It's close! But `StatelessClientOptions` does not have an easy way to set Headers so It will be missing the `apikey`.
I also found this https://github.com/supabase-community/gotrue-csharp where they talk about `StatelessClientOptions` as well.
However, passing options like demonstrated causes an error because of the missing `apikey` header.
Id like to be able to do something like this.
<img width="782" alt="Screenshot 2023-10-03 at 09 17 13" src="https://github.com/supabase-community/supabase-csharp/assets/1787881/f2894c24-56b9-4d91-911c-107774f90944">

**Comment by wiverson** [comment]
Is this not working?

https://github.com/supabase-community/gotrue-csharp/blob/a50dd2daaac989883be541aa57372e9f0eabb0fa/Gotrue/StatelessClient.cs#L258

**Comment by stopyellingatme** [comment, marked as answer]
Shoot, that did work! Thank you for your help and attention. I really appreciate it.
